vendor/bat-native-ads/src/bat/ads/internal/bundle.cc

```diff
@@ -188,6 +193,50 @@ std::unique_ptr<BundleState> Bundle::GenerateFromCatalog(
   return state;
 }
 
+bool Bundle::DoesOsSupportCreativeSet(
+    const CreativeSetInfo& creative_set) {
+  if (creative_set.oses.empty()) {
+    // Creative set supports all OSes
+    return true;
+  }
+
+  const std::string client_os = GetClientOS();
+  for (const auto& os : creative_set.oses) {
+    if (os.code == client_os) {
```

Comment:
@tmancey this might match against `os.name` rather than `os.code` judging by the example in https://github.com/brave/brave-browser/issues/5231:

`          "oses": [
            {
              "code": "i1g4cO6Pl",
              "name": "windows"
            }
          ],`

Reply:
Thanks I have updated code